features/storage/csi_clone.feature

```diff
@@ -0,0 +1,67 @@
+Feature: CSI clone testing related feature
+  """
+  This test can only work on a Cluster which has CSI driver deployed, OSP has csi deploy by default from 4.7. 
+  When deploy a cluster which has no csi deployed, please use enable_tpnu_featureset: "yes" in LAUNCHER_VARS.
+  From 4.7, OSP CSI driver was installed by default. Its storage class name is standard-csi.
+  Use below command can temporarily add cs class(oc get sc, can find all deployed stroage class):
+  oc patch featuregate cluster -p '{"spec": {"featureSet": "TechPreviewNoUpgrade"}}' --type merge
+  """
+  
+  # @author jianl@redhat.com
+  # @case_id OCP-27615
+  @admin
+  Scenario: Clone a pvc on OSP in the same namespace is successfully
+    Given the master version >= "4.8"
+    Given I switch to cluster admin pseudo user
+    # Step 1
+    # From 4.7, OSP CSI driver was installed by default. The storage class name is standard-csi
+    # Step 2
+    When admin creates a project with:
+      | project_name | clone |
+    Then the step should succeed
+    Given I use the "clone" project
+    When I obtain test data file "storage/csi/pod_pvc_sc.yaml"
+    And I run the :create client command with:
+      | f | pod_pvc_sc.yaml |
+    Then the step should succeed
+    And the pod named "mypod" becomes ready
+    And the "mypvc" PVC becomes :bound
+    # Step 3
+    Given 1 pod becomes ready with labels:
+      | name=frontendhttp |
+    And I wait for the steps to pass:
+    """
+    When I execute on the pod:
+      | sh | -c | touch /mnt/local/testfile |
+    Then the step should succeed
+    """
+    And I wait for the steps to pass:
+    """
+    When I execute on the pod:
+      | sh | -c | echo "clone test" > /mnt/local/testfile |
+    Then the step should succeed
+    """
```

Comment:
Line 32~42 can be shortened as.
```
And I wait for the steps to pass:
"""
When I execute on the pod:
      | sh | -c | echo "clone test" > /mnt/local/testfile |
Then the step should succeed
"""
```

The `echo` can create the content and the file if the file is not present, thus `touch` is not needed.

Reply:
In my impression you are correct and actually I tried it as you said, but unfortunately it doesn't work. It take me hours to fix it... @jhou1

Reply:
BTW, maybe the "And I wait for the steps to pass:" is not needed, we just execute cmd and the step should pass.

Reply:
@duanwei33 ok, i will try it, if it can pass stable, i will remove it.

Reply:
Where is the log for the failures?

Reply:
No error log in these steps, the latest step of verify testfile could not output "clone test" because there is no file cloned to the new pod.

Reply:
@JianLi-RH Do you mean that when only using the following steps,  the testfile is not written to clone pvc when checking? 
> When I execute on the pod:
>       | sh | -c | echo "clone test" > /mnt/local/testfile |
> Then the step should succeed

From my view:
1. "And I wait for the steps to pass:" This step could be removed safely, we don't use the waiting loop for a simple step which the result will not change.
2. When you get the fail, we are not sure if it fails because no data writing into origin pvc or it fails because the clone feature, could you update a bit as below, it will help us to catch what is not correct. 
> When I execute on the pod:
>       | sh | -c | echo "clone test" > /mnt/local/testfile; ls /mnt/local/testfile |
> Then the step should succeed

Reply:
@duanwei33 after failure, I logged into the origin pod and found the testfile is created with correct content, and then logged into the cloned pod I found there is no such file. I tried same step manually, they all pass. So I assume there is something delayed before creating the cloned pvc.
BTW, to run same code on my local machine, it is very stable. On Jenkins,  it is not stable.

Reply:
Hmm, from my view, if failed case run as you said, I prefer to treat it as a bug. I think you can update with our comments, our script is not to avoid the failure, and we can debug what caused that, of course maybe by PSI unstable, but it is acceptable. WDYT?

Reply:
I'm curious why that had failed too. Let's be more communication efficient. Next time we talk about failures, let's make sure we attach logs so that relevant owners can know what the underlying problem is. Logs are more helpful than the language here.

Reply:
@duanwei33 @jhou1 here is one failed run: [https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/ocp-common/job/Runner/47232/console](url),  you can see it could not find testfile on the clone pod:
0:34:23      When I execute on the pod:                                       # features/step_definitions/pod.rb:216
10:34:23        | sh | -c | more /mnt/local/testfile |
10:34:23        [02:34:22] INFO> Shell Commands: oc exec mypod-clone  --kubeconfig=/home/jenkins/ws/workspace/ocp-common/Runner/workdir/ocp4_testuser-0.kubeconfig -n sqkbk  -i -- sh -c more\ /mnt/local/testfile
10:34:23        
10:34:23        STDERR:
10:34:23        more: stat of /mnt/local/testfile failed: No such file or directory